pose_est_nets/utils/plotting_utils.py

```diff
@@ -351,94 +257,153 @@ def predict_videos(
-
-        # save csv file of predictions in DeepLabCut format
-        num_joints = model.num_keypoints
-        predictions = np.zeros((keypoints_np.shape[0], num_joints * 3))
-        predictions[:, 0] = np.arange(keypoints_np.shape[0])
-        # put x vals back in original pixel space
-        x_resize = cfg.data.image_resize_dims.width
-        x_og = cfg.data.image_orig_dims.width
-        predictions[:, 0::3] = keypoints_np[:, 0::2] / x_resize * x_og
-        # put y vals back in original pixel space
-        y_resize = cfg.data.image_resize_dims.height
-        y_og = cfg.data.image_orig_dims.height
-        predictions[:, 1::3] = keypoints_np[:, 1::2] / y_resize * y_og
-        predictions[:, 2::3] = confidence_np
-
-        # get bodypart names from labeled data csv if possible
-        if ("data_dir" in cfg.data) and ("csv_file" in cfg.data):
-            csv_file = os.path.join(cfg.data.data_dir, cfg.data.csv_file)
-        else:
-            csv_file = ""
-        if os.path.exists(csv_file):
-            if "header_rows" in cfg.data:
-                header_rows = list(cfg.data.header_rows)
-            else:
-                # assume dlc format
-                header_rows = [0, 1, 2]
-            df = pd.read_csv(csv_file, header=header_rows)
-            # collect marker names from multiindex header
-            joint_labels = [c[0] for c in df.columns[1::2]]
-        else:
-            joint_labels = ["bp_%i" % n for n in range(model.num_keypoints)]
-
-        # build data frame
-        xyl_labels = ["x", "y", "likelihood"]
-        pdindex = pd.MultiIndex.from_product(
-            [["%s_tracker" % cfg.model.model_type], joint_labels, xyl_labels],
-            names=["scorer", "bodyparts", "coords"],
+        make_predictions_and_create_csv(
+            cfg = cfg, 
+            model = model, 
+            dataloader = predict_loader,
+            n_frames_ = n_frames_,
+            batch_size = sequence_length, #note this is different from the batch_size defined above
+            save_file = save_file,
+            data_name =  video_file
         )
-        df = pd.DataFrame(predictions, columns=pdindex)
-        if save_file.endswith(".csv"):
-            df.to_csv(save_file)
-        elif save_file.find(".h") > -1:
-            df.to_hdf(save_file)
-        else:
-            raise NotImplementedError("Currently only .csv and .h5 files are supported")
+
 
     # if iterating over multiple models, outside this function, the below will reduce
     # memory
     del model, pipe, predict_loader
     torch.cuda.empty_cache()
+
+def make_predictions_and_create_csv(
+    cfg: DictConfig,
+    model: LightningModule,
+    dataloader: torch.utils.data.DataLoader,
+    n_frames_: int, #total number of frames,
+    batch_size: int, #batch size or sequence length
+    save_file: str,  
+    data_name: str = "dataset",
+    save_heatmaps: bool = False,
+):
+    keypoints_np = np.zeros((n_frames_, model.num_keypoints * 2))
```

Comment:
great refactoring. it is still a long function that can be further nicely refactored into three (still large) funcs -- 
1. `predict()`: lines `286-328`
2. `undo_resize()`: lines `330-342`
3.  `build_df_and_save()`: lines `344-373`

Reply:
made 1,2. still need to do 3